File: ui/src/types/discover.ts
export interface LeaderboardEntry {
  rank: string
  proxy_wallet: string
  user_name: string
  vol: number
  pnl: number
}

export interface TraderProfile {
  address: string
  trade_count: number
  top_category: string
  win_rate: number | null
}

export interface DiscoverState {
  leaderboard: LeaderboardEntry[]
  selected_profile: TraderProfile | null
  logs: { level: string; message: string }[]
}

File: src/commands/dto/discover.rs
//! Discover tab DTOs.

use serde::Serialize;

#[derive(Serialize, Clone)]
pub struct LeaderboardEntryDto {
    pub rank: String,
    pub proxy_wallet: String,
    pub user_name: String,
    pub vol: f64,
    pub pnl: f64,
}

#[derive(Serialize, Clone)]
pub struct TraderProfileDto {
    pub address: String,
    pub trade_count: u32,
    pub top_category: String,
    pub win_rate: Option<f64>,
}
